partner_company_default/models/res_company.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    @api.model
+    def create(self, vals):
+        self = self.with_context(creating_from_company=True)
+        company = super(ResCompany, self).create(vals)
+        # Add partner's company_id to new company
+        company.partner_id.sudo().write({"company_id": company.id})
```

Comment:
This is a potential problem. Companies' partner should have company_id = False. Why do you want to do that?

Reply:
Yes. I agree.